capa/rules/__init__.py

```diff
@@ -1280,11 +1280,6 @@ def __init__(
 
         ensure_rule_dependencies_are_met(rules)
 
-        if rules_filter_func:
```

Comment:
The addition of ruleset filtering might be premature optimization in the mean time anyways. So I suggest we think about measuring the performance difference first before prioritizing updating the cache and adding this back in.